lxd/instance/drivers/driver_qemu.go

```diff
@@ -677,6 +677,20 @@ func (d *qemu) ovmfPath() string {
 	return "/usr/share/OVMF"
 }
 
+func killQemuProcess(d *qemu, pid int) {
+	proc, err := os.FindProcess(pid)
+	if err != nil {
+		d.logger.Error("Failed to find VM process", log.Ctx{"pid": pid})
+		return
+	}
+
+	proc.Kill()
```

Comment:
looks like you haven't assign the return value of proc.Kill to err here

Reply:
We should also have a call to `proc.Wait()` here to return once the VM process has been killed.

https://golang.org/pkg/os/#Process.Wait

Reply:
Keep in mind that according to the Go docs, `os.FindProcess()` always returns a process on unix systems even if it doesn't exist, e.g. https://golang.org/pkg/os/#FindProcess

Reply:
> Keep in mind that according to the Go docs, os.FindProcess() always returns a process on unix systems even if it doesn't exist. 

Thanks for the heads up, I updated the code to skip the error check after `FindProcess` and inspect the error message after `proc.Kill()` to distinguish whether the process existed or not. 

> We should also have a call to proc.Wait() here to return once the VM process has been killed.

Added this, but I've been using the approach you described for simulating the qemu error condition (removing the monitor socket and restarting lxd) so it produces an error because the restarted lxd isn't the parent of the qemu process. I think under ordinary circumstances it probably would still be the parent.

Reply:
@s-alty ah true, I'd not thought about that, makes sense.

Although I didn't mean for you to remove the error checking for FindProcess, just be aware that it should always succeed, and instead use existence of the proc file as indicator the process exists still.

Reply:
Sure, I can add that FindProcess error check back.

I don't think the current implementation can ever throw one though:
https://github.com/golang/go/blob/1d5e14632edc2ba76156c8a771a2a1a5c5387326/src/os/exec_unix.go#L95-L98

Reply:
Thanks.